server/src/main/kotlin/suwayomi/tachidesk/server/ServerSetup.kt

```diff
@@ -236,14 +236,27 @@ fun applicationSetup() {
             serverConfig.socksProxyEnabled,
             serverConfig.socksProxyHost,
             serverConfig.socksProxyPort,
-        ) { proxyEnabled, proxyHost, proxyPort ->
-            Triple(proxyEnabled, proxyHost, proxyPort)
+            serverConfig.socksProxyUsername,
+            serverConfig.socksProxyPassword,
+        ) { proxyEnabled, proxyHost, proxyPort, proxyUsername, proxyPassword ->
+            data class DataClassForDestruction(
+                val proxyEnabled: Boolean,
+                val proxyHost: String,
+                val proxyPort: String,
+                val proxyUsername: String,
+                val proxyPassword: String,
+            )
+            DataClassForDestruction(proxyEnabled, proxyHost, proxyPort, proxyUsername, proxyPassword)
         },
```

Comment:
```suggestion
            serverConfig.socksProxyUsername,
            serverConfig.socksProxyPassword,
        ) { proxyEnabled, proxyHost, proxyPort, proxyUsername, proxyPassword ->
            data class ProxySettings(
                val proxyEnabled: Boolean,
                val proxyHost: String,
                val proxyPort: String,
                val proxyUsername: String,
                val proxyPassword: String,
            )
            ProxySettings(proxyEnabled, proxyHost, proxyPort, proxyUsername, proxyPassword)
        }.distinctUntilChanged(),
```

Reply:
API says it doesn't need `distinctUntilChanged()`
![image](https://github.com/Suwayomi/Suwayomi-Server/assets/8719233/13e747be-6856-408c-84e7-b7de87445988)

Reply:
Its just to be safe.